be/src/util/compression/stream_compression.cpp

```diff
@@ -422,6 +422,391 @@ Status ZstandardStreamCompression::decompress(uint8_t* input, size_t input_len,
+                        matchAddress -= decrement64;
+                    } else {
+                        COPY_VALUE(output, matchAddress, int64_t);
+                        matchAddress += SIZE_OF_LONG;
+                        output += SIZE_OF_LONG;
+                    }
+
+                    if (matchOutputLimit >= fastOutputLimit) {
+                        if (matchOutputLimit > outputLimit) {
+                            std::stringstream ss;
+                            ss << "LzoStreamCompression error: ";
+                            return Status::InternalError(ss.str());
+                        }
+
+                        while (output < fastOutputLimit) {
+                            COPY_VALUE(output, matchAddress, int64_t);
+                            matchAddress += SIZE_OF_LONG;
+                            output += SIZE_OF_LONG;
+                        }
+
+                        while (output < matchOutputLimit) {
+                            *(output++) = *(matchAddress++);
+                        }
+                    } else {
+                        while (output < matchOutputLimit) {
+                            COPY_VALUE(output, matchAddress, int64_t);
+                            matchAddress += SIZE_OF_LONG;
+                            output += SIZE_OF_LONG;
+                        }
+                    }
+                }
+                output = matchOutputLimit; // correction in case we over-copied
+            }
+
+            // copy literal
+            uint8_t* literalOutputLimit = output + literalLength;
+            if (literalOutputLimit > fastOutputLimit || input + literalLength > inputLimit - SIZE_OF_LONG) {
+                if (literalOutputLimit > outputLimit) {
+                    std::stringstream ss;
+                    ss << "LzoStreamCompression error: ";
+                    return Status::InternalError(ss.str());
+                }
+
+                // slow, precise copy
+                memcpy(output, input, literalLength);
+                input += literalLength;
+                output += literalLength;
+            } else {
+                // fast copy. We may over-copy but there's enough room in input
+                // and output to not overrun them
+                do {
+                    COPY_VALUE(output, input, int64_t);
+                    input += SIZE_OF_LONG;
+                    output += SIZE_OF_LONG;
+                } while (output < literalOutputLimit);
+                // adjust index if we over-copied
+                input -= (output - literalOutputLimit);
+                output = literalOutputLimit;
+            }
+            lastLiteralLength = literalLength;
+        }
+
+        if (input + SIZE_OF_SHORT > inputLimit && *reinterpret_cast<const int16_t*>(input) != 0) {
+            std::stringstream ss;
+            ss << "LzoStreamCompression error: ";
+            return Status::InternalError(ss.str());
```

Comment:
No need to do it, just 

> Status::internalError("lzo stream decompression error");

Reply:
another suggestion, is to include __line__ information.

it could be like

> return Status::InternalError(fmt::format("Compresstipon error at line {}", __line__));

Reply:
> another suggestion, is to include **line** information.
> 
> it could be like
> 
> > return Status::InternalError(fmt::format("Compresstipon error at line {}", **line**));

OK, I recommit the code, Please review